tests/Dfe.PlanTech.Web.E2ETests/cypress/e2e/dynamic-page-validator/validators/quick-questions.js

```diff
@@ -0,0 +1,23 @@
+import { CleanText } from "../helpers/index.js";
+
+export const quickNavigateQuestions = (path, section) => {
```

Comment:
Why has this been named `quickNavigate`? same for the one above? it looks like it just skips two of the checks, which I vaguely recall us talking about - as a "can skip because its probably overkill to do for all pages"

Though, as its only two checks, how much of a speed increase is this?
I reckon either the existing question validation should be used _or_ the stuff in this file can be a shared method thats used in both places, as this is duplicating some of the question.js code

Reply:
or maybe simpler, have a parameter in the current `questions.js` validator for `checkAnswers` and only validate the answers if that is set to true, and don't do that here

Reply:
We have answer validation in the all-questions paths in dpv1 (to be renamed). The quick versions skips two checks but for each questions, it doesn't save massive amounts of time, it's more about cutting down the number of things that could fail and stop us getting to the recommendations. As it stands if we have a typo in the helptext (for example) then we can't complete the path and can't get to the recommendations at all. That would now only fail on the test that's designed for it. Likewise with the section stuff, interstitial page etc.
I'll consider a parameter but if it doesn't work on my first try I'm going to stick it in the "random improvements" ticket I've got a bunch of notes on, because cypress is unreasonably finicky about conditional testing and I can see it sucking me in for an unreasonable length of time...

Reply:
Oh interesting, what happens when you test something within an if condition statement?

Reply:
Ok I've done a bit more reading and I'd misremembered why it was unhappy with conditional testing I'd tried to do before, it was to do with not having conditions that relate to the state of the DOM, which wouldn't be an issue here. I've refactored and it works, but involves passing a boolean so I'm not a fan and I think it'll upset Sonarcloud. Were you thinking of another way?

Reply:
does it not like it simply as its conditionally choosing whether or not to run some test code? whats the error? Just ideally if possible I think it would be better if we can avoid having two methods that are the same except for one skipping some validation, maybe a validator method could be passed in that runs which might appease sonar cloud? like what the section validation is doing

Reply:
It's ok with this scenario, it's just when you're relying on what could be an unstable DOM. They've got a whole section in the docs on it that's quite good [here](https://docs.cypress.io/guides/core-concepts/conditional-testing). Anyway this isn't an issue for us.
I haven't committed it yet so not got the Sonarcloud results but I imagine it'll be [this rule](https://rules.sonarsource.com/javascript/type/Code%20Smell/RSPEC-2301/) that it's breaking. I'll do a commit to test my theory :)
Section validation is in two new functions now (and these ones have test cases in them so it's messy) so the original validateSection or whatever it's now called doesn't use a validator function any more. I'm trying to wrap my head around this idea... in the way we were doing validateSection, it was running the section stuff then if we were doing recommendations (ie if there was a validator fn) it would run the recommendations functions. For this one, we call (eg) doQuestions and pass a validator if we want the extra (?) tests, then call the validator inside the loop in doQuestions to call it for each q?

Reply:
Ah yep that sure does look like it would hit it, nicely spotted, I don't know if sonarclouds lesser rules for dev will hit it but I think you're right, the ones in main would. 

I just think there must be some way to avoid the duplication, so with a validator it would be something like
```js
export const navigateAndValidateQuestionPages = (path, section, extraValidator = None) => {
...
if (extraValidator)
    extraValidator(matchingQuestion)
...
}
```
but feel free to argue back that its worse, has its own cons. I'm not sure if I like it or not
going bigger picture again

> As it stands if we have a typo in the helptext (for example) then we can't complete the path and can't get to the recommendations at all

how likely is this in general once we've smoothed out dpv errors and its up and running day to day?